dataland-csvconverter/src/main/kotlin/org/dataland/csvconverter/csv/CsvUtils.kt

```diff
@@ -32,36 +30,88 @@ object CsvUtils {
      * This function uses the backing map to extract a property from a CSV row.
      * If the requested column is not populated "n/a" will be returned
      */
-    fun Map<String, String>.getCsvValue(property: String, csvData: Map<String, String>): String {
-        return csvData[this[property]!!]!!.trim().ifBlank {
-            NOT_AVAILABLE_STRING
+    fun Map<String, String>.getCsvValue(property: String, csvData: Map<String, String>): String? {
+        return csvData[this[property]!!.lowercase()]?.trim()?.ifBlank {
+            null
         }
     }
 
+    /** This function checks, whether the passed properties field contains an entry.
+     * If it does, it returns true
+     */
+    fun Map<String, String>.checkIfFieldHasValue(property: String, csvData: Map<String, String>): Boolean {
+        return csvData[this[property]!!.lowercase()]!!.isNotBlank()
+    }
+
     /**
-     * This function uses the backing map to extract a numeric property from a CSV row.
-     * The returned value is scaled according to the scaleFactor
+     * Checks if any of the specified fields have a value
      */
-    fun Map<String, String>.getScaledCsvValue(
+    fun Map<String, String>.checkIfAnyFieldHasValue(fields: List<String>, csvData: Map<String, String>): Boolean {
+        return fields.any { this.checkIfFieldHasValue(it, csvData) }
+    }
+
+    /**
+     * Tries to parse a percentage value in the format "XX,XX %" with a comma separator
+     */
+    fun Map<String, String>.readCsvPercentage(property: String, csvData: Map<String, String>): BigDecimal? {
+        val rawValue = this.getCsvValue(property, csvData)?.trim() ?: return null
+        val expectedFormat = "\\d+(,\\d+)?(\\s*)%".toRegex()
+        if (!rawValue.matches(expectedFormat))
+            throw IllegalArgumentException(
+                "The input string \"$rawValue\" for column ${this[property]} does not " +
+                    "match the expected format for a percentage value"
+            )
+
+        return rawValue
+            .replace("[%\\s]".toRegex(), "")
+            .replace(",", ".")
+            .toBigDecimal()
+            .multiply("0.01".toBigDecimal())
+            .stripTrailingZeros()
+    }
+
+    /**
+     * Tries to parse a decimal value from the CSV file with the expected format
+     * XXX.XXX,XXX
+     */
+    fun Map<String, String>.readCsvDecimal(
         property: String,
         csvData: Map<String, String>,
-        scaleFactor: String
+        scaleFactor: BigDecimal = BigDecimal.ONE
```

Comment:
Ich dachte der Skalenfaktor kann jetzt weg?

Reply:
Jein. Also z.B. das Total CapEx field ist im CSV in Millionen angegeben, aber im Datenmodell normal. Vorher hatten wir allerdings eine impliziete Konvertierung in der hauptsächlich benutzten Funktion (einfach alles 1000000). Die fällt jetzt weg und man muss ein scaling für jeden Datenpunkt angeben. (Altes hat auch dafür gesorgt, dass Prozentzahlen, die ohne % angegeben wurden mit 1000000 anstelle von 0.01 multipliziert wurden)